docs/Developers/Design/crd_design.md

```diff
@@ -0,0 +1,58 @@
+# Custom Resource Definitions
+
+## **StorageCatalog**
+
+### *Singular: StorageCatalog, Plural: StorageCatalogs*
+
+```YAML
+kind: StorageCatalog
+spec:
+  - storage: "ceph"
+    version: ["mimic", "nautilus"]
+    alert: true
+    dashboard: true
+```
+
+## **StorageAlert**
+
+### *Singular: StorageAlert, Plural: StorageAlerts*
+
+```YAML
+kind: StorageAlert
+spec:
+  - storage: "ceph"
+    - version: "mimic"
+      alert: false
+    - version: "nautilus"
+      alert: true
+      prometheus:
+      - namespace: "monitoring"
+      - label: "prometheus:k8s"
+```
+
+## **StorageDashboard**
+
+### *Singular: StorageDashboard, Plural: StorageDashboards*
+
+```YAML
+kind: StorageDashboard
+spec:
+  - storage: "ceph"
+    - version: "mimic"
+      dashboard: false
+    - version: "nautilus"
+      dashboard: true
+      grafana:
+      - namespace: "monitoring"
+      - label: "grafana:k8s"
+```
+
+## **StorageMonitoringPlan**
```

Comment:
What is this meant for?

Reply:
CRD that the operator will orchestrate against.
Consider you have two clusters. In 1st cluster you have nautilus running and want to enable alerting, in 2nd cluster you have some other version and don't want alerting enabled.
This means operator would have 2 desired states to orchestrate against. So, if we have this CRD, we could ask operator to orchestrate this CRD itself and defer state orchestration to respective controller. Also, by default this would be initialized by the operator itself and not the user/admin.